src/foam/core/stdlib.js

```diff
@@ -276,8 +276,8 @@ foam.LIB({
           match = str.match(/^(async )?function(\s+[_$\w]+|\s*)\((.*?)\)/);
 
       if ( ! match ) {
-        /* istanbul ignore next */
-        throw new TypeError("foam.Function.argsStr could not parse input function:\n" + ( f ? f.toString() : 'undefined' ) );
+        console.warn("foam.Function.argsStr could not parse input function:\n" + ( f ? f.toString() : 'undefined' ) );
```

Comment:
I think it would be better to preserve the function and catch+log the warning at usage spots like AbstractMethod.args.factory

Reply:
Good call. Done.

Reply:
Ah missed a spot where this will break in debug type checking. Fixed there too.